modules/Movecraft/src/main/java/net/countercraft/movecraft/processing/tasks/detection/SubcraftValidator.java

```diff
@@ -0,0 +1,53 @@
+package net.countercraft.movecraft.processing.tasks.detection;
+
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.craft.CraftManager;
+import net.countercraft.movecraft.craft.CraftType;
+import net.countercraft.movecraft.localisation.I18nSupport;
+import net.countercraft.movecraft.processing.IMovecraftWorld;
+import net.countercraft.movecraft.processing.MovecraftWorld;
+import net.countercraft.movecraft.processing.TaskPredicate;
+import net.countercraft.movecraft.util.ChatUtils;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class SubcraftValidator implements TaskPredicate<MovecraftLocation> {
+    @Override
+    public Result validate(@NotNull MovecraftLocation movecraftLocation, @NotNull CraftType type, @NotNull MovecraftWorld world, @Nullable CommandSender player) {
+
+        if (!type.getMustBeSubcraft()) {
+            return Result.succeed();
+        }
+        IMovecraftWorld iMovecraftWorld;
+        try {
+            iMovecraftWorld = (IMovecraftWorld) world;
+        } catch (ClassCastException e) {
+            e.printStackTrace();
+            return Result.failWithMessage(ChatUtils.ERROR_PREFIX+ " An invalid world was passed to a validator!");
+        }
```

Comment:
You shouldn't use a try catch here, but rather an `instanceof` expression. Furthermore, why are you even casting to the concrete class? You're not using it anywhere.

Reply:
The concrete class is necessary here because I use ``IMovecraftWorld.equals()`` which is not defined for the abstract class.

Reply:
all objects implement equals, notice how you never use `iMovecraftWorld` and your code still functions

Reply:
wait, i just noticed this :woman_facepalming: my apologies